server/controllers/transactions/update_state.js

```diff
@@ -17,21 +17,39 @@ const sanitization = {
 module.exports = (req, res) => {
   sanitize(req, res, sanitization)
   .then(params => {
-    const { transaction, state } = req.body
-    const reqUserId = req.user._id
-    return transactions_.byId(transaction)
-    .then(VerifyRights(state, reqUserId))
-    .then(transactions_.updateState.bind(null, state, reqUserId))
-    .then(Track(req, [ 'transaction', 'update', state ]))
+    return updateState(params)
+    .then(Track(req, [ 'transaction', 'update', params.state ]))
   })
   .then(responses_.Ok(res))
   .catch(error_.Handler(req, res))
 }
 
-const VerifyRights = (state, reqUserId) => {
+const updateState = async ({ transactionId, state, reqUserId }) => {
+  const transaction = await transactions_.byId(transactionId)
+  verifyRights(transaction, state, reqUserId)
```

Comment:
```suggestion
  validateAccessLevel(transaction, state, reqUserId)
```
or
```suggestion
  validateRights(transaction, state, reqUserId)
```

Reply:
done in a47b732: I went for option 2, as access level might be confused with the access level derived from users' roles